Yoast/yoast-components
# Pass meta description placeholder to editor

## Summary

This PR can be summarized in the following changelog entry:

* Adds a meta description placeholder text

## Relevant technical choices:

* Adds a descriptionEditorFieldPlaceholder prop to the snippet editor that is passed along to the draftJS editor (the name of the variable is very long and can be changed but was specified in this way in order to avoid confusion with the already existing variable `descriptionPlaceholder`)
* The color of the placeholder text is currently a lighter shade of grey compared to how it is in the old snippet editor. An [issue](https://github.com/Yoast/wordpress-seo/issues/9874) has been created for that.

## Test instructions

This PR can be tested by following these steps:

* Empty your title and meta description templates in SEO > Search Appearance > Content Types
* Go to the snippet editor, edit snippet, and empty the meta description field
* There should be a placeholder text that says "Modify your meta description by editing it right here"
* Changing focus from the meta description field should cause the text color of the placeholder to be a darker grey
* Empty the title field and save the post/page
* The text in the title field should be %%title%% - %%sitename%%
* The title field text should act like normal text, i.e. be editable
* Test these functionalities for both posts and pages

Fixes https://github.com/Yoast/wordpress-seo/issues/9830
